### PR TITLE
[Hubs] Add support for compressed exports from Cost Management

### LINKED ISSUE #903: [Hubs] Add support for compressed exports from Cost Management
## 📝 Scenario
As a **FinOps practitioner**, I need to **ingest compressed exports from Cost Management** in order to **minimize my storage and processing costs**

## 💎 Solution
- Update managed exports to use parquet and gzipped files
- Update the ingestion pipeline to pull parquet and gzipped files in addition to CSV files

## 🙋‍♀️ Ask for the community

We could use your help:
1. Please vote this issue up (👍) to prioritize it.
2. Leave comments to help us solidify the vision.

### PR DESCRIPTION
**Pull request by MSBrett**
## 🛠️ Description
This PR adds support for ingesting native parquet files from Azure Cost Management.

1. Fully parameterize getExportBody function.
2. Update managed exports to emit parquet files by default.
3. Add parquet dataset for msexports container.
4. Update the msexports_etl_ingestion pipeline to pull parquet files in addition to CSV file.

This PR does not include support for .csv.gz files
Docs for self-managedexports will be updated in a separate PR.

Fixes #903 

### 🔬 How did you test this change?

> - [x] 🤏 Lint tests
> - [ ] 🤞 PS -WhatIf / az validate
> - [x] 👍 Manually deployed + verified
> - [ ] 💪 Unit tests
> - [ ] 🙌 Integration tests

### 🙋‍♀️ Do any of the following that apply?

> - [ ] 🚨 This is a breaking change.
> - [ ] 🤏 The change is less than 20 lines of code.

### 📑 Did you update `docs/changelog.md`?

> - [x] ✅ Updated changelog (required for `dev` PRs)
> - [ ] ➡️ Will add log in a future PR (feature branch PRs only)
> - [ ] ❎ Log not needed (small/internal change)

### 📖 Did you update documentation?

> - [ ] ✅ Public docs in `docs` (required for `dev`)
> - [ ] ✅ Internal dev docs in `src` (required for `dev`)
> - [ ] ➡️ Will add docs in a future PR (feature branch PRs only)
> - [x] ❎ Docs not needed (small/internal change)

### CODE REVIEW

File: src/templates/finops-hub/modules/dataFactory.bicep [hunk 1]
```diff
@@ -47,7 +47,7 @@ var ftkVersion = loadTextContent('ftkver.txt')
...
...a string, dataOverwriteBehavior string) string => '{ "properties": { "definition": { "dataSet": { "configuration": { "dataVersion": "${schemaVersion}", "filters": [] }, "granularity": "Daily" }, "timeframe": "${isMonthly ? 'TheLastMonth': 'MonthToDate' }", "type": "${datasetType}" }, "deliveryInfo": { "destination": { "container": "${exportContainerName}", "rootFolderPath": "@{if(startswith(item().scope, \'/\'), substring(item().scope, 1, sub(length(item().scope), 1)) ,item().scope)}", "type": "AzureBlob", "resourceId": "@{variables(\'storageAccountId\')}" } }, "schedule": { "recurrence": "${ isMonthly ? 'Monthly' : 'Daily'}", "recurrencePeriod": { "from": "2024-01-01T00:00:00.000Z", "to": "2050-02-01T00:00:00.000Z" }, "status": "Inactive" }, "format": "${exportFormat}", "partitionData": "${partitionData}", "dataOverwriteBehavior": "${dataOverwriteBehavior}", "compressionMode": "${compressionMode}" }, "id": "@{variables(\'resourceManagementUri\')}@{item().scope}/providers/Microsoft.CostManagement/exports/@{variables(\'exportName\')}", "name": "@{variables(\'exportName\')}", "type": "Microsoft.CostManagement/reports", "identity": { "type": "systemAssigned" }, "location": "global" }'
```
Comment (flanakin) [inline src/templates/finops-hub/modules/dataFactory.bicep:50]: Why do we need to make format, compression, partitioning, and overwrite inputs? Why not force those based on the scenario?

File: src/templates/finops-hub/modules/dataFactory.bicep [hunk 2]
```diff
@@ -376,6 +376,35 @@ resource dataset_msexports 'Microsoft.DataFactory/factories/datasets@2018-06-01'
...
+resource dataset_msexports_parquet 'Microsoft.DataFactory/factories/datasets@2018-06-01' = {
+  name: '${safeExportContainerName}_parquet'
+  parent: dataFactory
+  properties: {
+    annotations: []
+    parameters: {
+      blobPath: {
+        type: 'String'
+      }
+      }
```
Comment (flanakin) [inline src/templates/finops-hub/modules/dataFactory.bicep:388]:
```suggestion
    }
```

File: src/templates/finops-hub/modules/dataFactory.bicep [hunk 3]
```diff
@@ -1354,7 +1383,7 @@ resource pipeline_ConfigureExports 'Microsoft.DataFactory/factories/pipelines@20
                 }
                 method: 'PUT'
                 body: {
-                  value: getExportBody(exportContainerName, focusSchemaVersion, true)
+                  value: getExportBody(exportContainerName, 'FocusCost', focusSchemaVersion, false, 'Parquet', 'Snappy', 'true', 'CreateNewReport')
```
Comment (flanakin) [inline src/templates/finops-hub/modules/dataFactory.bicep:1386]: I believe the isMonthly flag should still be `true`...
```suggestion
                  value: getExportBody(exportContainerName, 'FocusCost', focusSchemaVersion, true, 'Parquet', 'Snappy', 'true', 'CreateNewReport')
```

File: src/templates/finops-hub/modules/dataFactory.bicep
Comment (flanakin) [inline src/templates/finops-hub/modules/dataFactory.bicep:1]: Don't forget to update the changelog!

**Review by flanakin — APPROVED** [review]

File: src/templates/finops-hub/modules/dataFactory.bicep (hunk 1 above)
Comment (MSBrett) [reply src/templates/finops-hub/modules/dataFactory.bicep:50]: This is groundwork to allow one to specify the type of file generated by CM exports for scenarios where parquet isn't appropriate.

File: src/templates/finops-hub/modules/dataFactory.bicep
Comment (MSBrett) [reply src/templates/finops-hub/modules/dataFactory.bicep:1]: Done!

**Pull request merged**